### PR TITLE
Nested Recipes-Shotgun Ammo + add missing difficulty field

### PR DESCRIPTION
**Pull request by TheShadowFerret**
#### Summary
Interface "Nested recipes for shotgun ammunition"

#### Purpose of change

Further nesting, also I forgot to add the difficulty field to #60167.

#### Describe the solution

Nest 12 gauge, 20ga percussion, blunderbuss, and both 40mm types into categories, the last subnesting into x46 and x53mm. .410 and signal flare were left out due to only one recipe for those calibers.

#### Describe alternatives you've considered

~~Make ANickelN do it~~

#### Testing

image
Ensure no recipes were lost from nesting.
#### Additional context

~~Why do the id suffixes for using a dowel have a space 😡~~

### CODE REVIEW

**Comment by ANickelN** [comment]
Talking with bombasticSlacks, he has a PR coming in very soon that will make things autolearn at the appropriate time instead of having to be set manually.  

He instructed to remove "autolearn" from any new nested categories I add, as they will throw errors in the new system, and then set my PR as a draft or leave a note that it should be merged after his PR.

https://github.com/CleverRaven/Cataclysm-DDA/pull/60302

**Comment by ANickelN** [comment]
It's failing because two recipes are not able to be pathed to and need to be added to the nest:
 blun_flechette
blun_slug_using_lead

```
(~[slow] ~[.],starting_items)=> (continued from above) ERROR : src/recipe_dictionary.cpp:609 [static void recipe_dictionary::finalize()] recipe blun_flechette is nested but isn't in a nested group.  It will be impossible to path to by the player.
(~[slow] ~[.],starting_items)=> 23:43:07.085 ERROR : src/recipe_dictionary.cpp:609 [static void recipe_dictionary::finalize()] recipe blun_slug_using_lead is nested but isn't in a nested group.  It will be impossible to path to by the player.
```

**Comment by tenmillimaster** [comment]
> Why do the id suffixes for using a dowel have a space 

Sorryyyyyyyyy

**Comment by TheShadowFerret** [comment]
I fixed the earlier merge conflicts, don't forget me before 0.G-eeeee 🙏

File: data/json/recipes/ammo/40x46mm.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
     "result": "40x46mm_buckshot_m118",
     "type": "recipe",
     "activity_level": "LIGHT_EXERCISE",
-    "category": "CC_AMMO",
-    "subcategory": "CSC_AMMO_SHOT",
+    "category": "CC_*",
```
Comment (Maleclypse) [inline data/json/recipes/ammo/40x46mm.json:6]:
```suggestion
    "category": "CC_*"
```
Resetting tests

File: data/json/recipes/ammo/40x46mm.json [hunk 2]
```diff
@@ -3,8 +3,8 @@
     "result": "40x46mm_buckshot_m118",
     "type": "recipe",
     "activity_level": "LIGHT_EXERCISE",
-    "category": "CC_AMMO",
-    "subcategory": "CSC_AMMO_SHOT",
+    "category": "CC_*"
```
Comment (Maleclypse) [inline data/json/recipes/ammo/40x46mm.json:6]:
```suggestion
    "category": "CC_*",
```

**Comment by Maleclypse** [comment]
IMG_1015
Saw some errors on the rerun tests that look maybe related.

**Comment by Maleclypse** [comment]
Still has those same errors.

**Pull request closed without merging**